src/components/views/rooms/wysiwyg_composer/message.ts

```diff
@@ -0,0 +1,190 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Composer as ComposerEvent } from "@matrix-org/analytics-events/types/typescript/Composer";
+import { IContent, IEventRelation, MatrixEvent } from 'matrix-js-sdk/src/models/event';
+import { MatrixClient } from "matrix-js-sdk/src/matrix";
+import { THREAD_RELATION_TYPE } from "matrix-js-sdk/src/models/thread";
+
+import { PosthogAnalytics } from "../../../../PosthogAnalytics";
+import SettingsStore from "../../../../settings/SettingsStore";
+import { decorateStartSendingTime, sendRoundTripMetric } from "../../../../sendTimePerformanceMetrics";
+import { attachRelation } from "../SendMessageComposer";
+import { RoomPermalinkCreator } from "../../../../utils/permalinks/Permalinks";
+import { doMaybeLocalRoomAction } from "../../../../utils/local-room";
+import { CHAT_EFFECTS } from "../../../../effects";
+import { containsEmoji } from "../../../../effects/utils";
+import { IRoomState } from "../../../structures/RoomView";
+import dis from '../../../../dispatcher/dispatcher';
+
+interface SendMessageParams {
+    mxClient: MatrixClient;
+    relation?: IEventRelation;
+    replyToEvent?: MatrixEvent;
+    roomContext: IRoomState;
+    permalinkCreator: RoomPermalinkCreator;
+    includeReplyLegacyFallback?: boolean;
+}
+
+// exported for tests
+export function createMessageContent(
```

Comment:
This looks really similar to what the current composer does. Why is there a need to create this new function. In theory could we use the same as before or are there some slight changes in functionality?

Reply:
The function in the `SendMessageComposer` is a class function which is using directly class attributes.
I removed all the class adherences here. For example, we do not use the  `model` of the current composer, we already have one internally (the rust one).

Since the new version is more *functional*, I can use it in the current composer to avoid duplicate code but the wysiwyg will replace the composer, so I'm not convinced to add changes in the current one.